dagql/idproto/id.proto

```diff
@@ -12,8 +12,10 @@ option go_package = "./idproto";
 // objects. Alternatively it may be stored in a database and referred to via an
 // RFC-6920 ni://sha-256;... URI.
 message ID {
-  // The parent ID, if any.
-  ID parent = 1;
+  // The ID of the object that the field selection will be evaluated against.
+  //
+  // If not present, the root Query object is implied.
+  ID base = 1;
```

Comment:
This was renamed from "parent" because it was too confusing having "parent" in the otel sense (parent span/vertex) and "parent" in the layering sense (chaining field selections to form an ID). "Base" feels more accurate here.